File: 1.Arrays/2.merge_sort.py
def merge_sorted_arrays(arr1, arr2):
    if len(arr1) == 0:
        return arr2
    if len(arr2) == 0:
        return arr1
    merged_array = []
    i = 0
    j = 0
    while i < len(arr1) and j < len(arr2):
        if arr2[j] > arr1[i]:
            merged_array.append(arr1[i])
            i += 1
        else:
            merged_array.append(arr2[j])
            j += 1
    while i < len(arr1):
        merged_array.append(arr1[i])
        i += 1
    while j < len(arr2):
        merged_array.append(arr2[j])
        j += 1
    return merged_array

print(merge_sorted_arrays([0,3,4,31], [3,4,6,30]))

File: 1.Arrays/2.mergeSort.js
function mergeSortedArrays(array1, array2){
    if(array1.length === 0) {
      return array2;
    }
    if(array2.length === 0) {
      return array1;
    }
    const mergedArray = [];
    let i = 0;
    let j = 0;    
  
    while (i < array1.length || j < array2.length){
     if(array2[j] == undefined || array1[i] <= array2[j]){
       mergedArray.push(array1[i]);
       i++;
     } else {
        mergedArray.push(array2[j]);
        j++;
      }
    }
    return mergedArray;
  }
  
  console.log(mergeSortedArrays([0,3,4,31], [3,4,6,30]));